projects/erp-hgrid/src/components/erp-hierarchical-grid.ts

```diff
@@ -0,0 +1,609 @@
+          return "danger";
+        case this.orderStatus.DELIVERED:
+          return "success";
+        default:
+            return "primary";
+    }
+  };
+
+  private getOrderStatusIconName = (status: string): string => {
+    switch (status) {
+        case this.orderStatus.PACKED:
+            return "dropbox";
+        case this.orderStatus.IN_TRANSIT:
+          return "delivery";
+        case this.orderStatus.CUSTOMS:
+          return "bill-paid";
+        case this.orderStatus.DELIVERED:
+          return "check";
+        default:
+            return "dropbox";
+    }
+  };
+
+  private statusTemplate = (ctx: IgcCellTemplateContext) => {
+    const cellValue: string = ctx.cell.value;
+    const badgeVariant: BadgeVariant = this.getOrderStatusBadgeVariant(cellValue);
+    const iconName: string = this.getOrderStatusIconName(cellValue);
+
+    return html`
+      <div class="status-cell">
+          <span>
+          <igc-badge
+            variant="${badgeVariant}"
+            shape="rounded">
+            <igc-icon name="${iconName}" collection="material" class="custom-icon"></igc-icon>
+          </igc-badge>
+          </span>
+          <span>${cellValue}</span>
+      </div>
+    `;
+  };
+
+  private countryTemplate = (ctx: IgcCellTemplateContext) => {
+    const cellValue: string = ctx.cell.value;
+    const flagPath: string = `country-flags/${cellValue}.svg`;
+
+    return html`
+      <div class="country-cell">
+          <span class="cup">
+              <img .src="${flagPath}"/>
+          </span>
+          <span>${cellValue}</span>
+      </div>
+    `;
+  };
+
+  // FORMATTERS
+  private formatNumberAsIs = (value: number): number => {
+    // Bypassing the default formatting of larger numbers
+    // Example for 4-digit numbers: 1,234 => 1234
+    return value;
+  }
+
+  private formatDate = (value: string): string => {
+    return value || 'N/A';
+  }
+
+  private formatAddress = (value: OrderDetails): string =>  {
+    return `${value.streetName} ${value.streetNumber}`;
+  }
+
+  private formatFullAddress = (value: OrderDetails): string => {
+    return `${value.streetNumber} ${value.streetName}, ${value.zipCode} ${value.city}, ${value.country}`;
+  }
+  
+  // PRODUCT IMAGE COLUMN OVERLAYS
+  private showTooltip = (event: MouseEvent, context: IgcCellTemplateContext): void => {
+    const targetEl: HTMLElement = event.target as HTMLElement;
+    const imageUrl: string = context.implicit;
+
+    // Set current hovered image properties
+    this.hoveredImageUrl = imageUrl;
+    const productName: string = context.cell.row?.cells?.find((c: any) => c.column.field === 'productName')?.value;
+    this.hoveredImageProductName = productName;
+
+    // Add floating-ui setups and styles
+    this.setupImageDialog(targetEl, this.productImageTooltip);
+    this.requestUpdate();
+  }
+
+  private setupImageDialog = (targetElement: HTMLElement, tooltip: HTMLElement): void => {
+    computePosition(targetElement, tooltip, {
```

Comment:
Can you add some delay when opening the image dialog so that it does not flicker when moving fast between the cells of this column: related to this: https://github.com/IgniteUI/grid-demos/issues/87